aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -59,12 +65,18 @@ index_select_add(const Tensor &select_indices,
                              const Tensor &src,
                              Tensor &output,
                              const Tensor& /*offsets*/,
-                             bool /*include_last_offset*/) {
+                             bool /*include_last_offset*/,
+                             index_t padding_idx,
+                             Tensor &bag_size) {
```

Comment:
I'm passing `bag_size` as a non-const reference (here and a few other places) because any time we encounter a padding index, we need to decrement the size for that bag. This might be bad design though. We could instead start with `bag_size` filled with zeros (in all the functions similar to `index_select_add`), and then increment when appropriate.